File: springboot-02-ioccontainer/src/main/java/com/linkedbear/springboot/assemble/a_module/config/BarConfiguration.java
package com.linkedbear.springboot.assemble.a_b_c_module_profile_conditional.config;

import com.linkedbear.springboot.assemble.a_b_c_module_profile_conditional.component.Bar;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class BarConfiguration {
    
    @Bean
    public Bar bbbar() {
        return new Bar();
    }
}

File: springboot-02-ioccontainer/src/main/kotlin/com/linkedbear/springboot/assemble/a_module/config/BarConfiguration.kt
package com.linkedbear.springboot.assemble.a_module.config

import com.linkedbear.springboot.assemble.a_module.component.Bar
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
open class BarConfiguration {
    @Bean
    open fun bbbar(): Bar {
        return Bar()
    }
}